mlflow/azure/client.py

```diff
@@ -146,3 +221,83 @@ def _is_valid_put_block_header(header_name):
         "x-ms-encryption-key-sha256",
         "x-ms-encryption-algorithm",
     }
+
+
+def _is_valid_adls_put_header(header_name):
```

Comment:
Do ADLS APIs reject requests with unrecognized headers? If not, do we need to validate?

Reply:
I tend to lean toward not sending unneeded info over the wire.  So while it may or may not throw, I'd rather not be yolo'ing out every header that somehow makes it into the list?

Reply:
I think that new headers are introduced rarely, so it seems to be fine.